### PR TITLE
Move Frontend to a Framework, and redesign for mobile

### PR DESCRIPTION
**Pull request by tfpk**
This is an intermediate PR to get a sense of if I'm going in the right direction. I'd welcome feedback if there's changes to the UX you think are really important.

I know it's not perfect -- some of the alignment needs fixing; but I think it's a good impression of what the final result would be.

### CODE REVIEW

**Comment by tfpk** [comment]
This is what it looks like, without expanding into the settings:
image

This is what it looks like with settings open:
image

This is it in mobile:
image


When in mobile, the settings panel hides the chat panel (the alternative would be unweildy):
image

**Comment by cyphar** [comment]
I'm not a huge fan of the desktop version to be honest, mainly because it feels like a stretched out tablet application (maybe the margins should be bigger -- and the buttons for "listen" and "send" probably shouldn't use the Android button icons on non-mobile devices). And "add a bot" should probably be a nice button rather than looking like a link. But the mobile version seems quite reasonable to me (though the message field and buttons seem like they have too much white space with the buttons feeling a bit small).

As for functionality, I think it would be nice to be able to collapse bots (`<details>` style) but we can do that as part of the #38 rework of bot management.

**Comment by tfpk** [comment]
I agree the desktop interface is very mobile-y. I am not convinced this is entirely bad -- consistent UI is good (and on those grounds I feel like the use of material icons is good on desktop). I suspect what would make this feel better is if messages on desktop were not full-width, but aligned so that your messages were right-aligned (and floating right) and other messages were left-aligned (and floating left). Alternatively, I agree that a larger margin would be desirable. I'll play with that tomorrow.

Agreed that minor UI fixes are definitely needed, they were just painful, and I didn't want to go to the effort without a general "this looks ok".

**Comment by kennib** [comment]
Aside from the issues with the messages. It looks pretty good to me. I only have a couple of changes that I would suggest before merging the PR.

The shadows of the message boxes are a bit distracting. They're drawing my eyes away from the message text. This is important to fix for the project videos.

- [ ] Make the message boxes more subtle (look at slack or MS teams for inspiration)

For the project video most of the action is happening in the new message box.
Can you make this a bit more prominent so people's eyes will be drawn there in videos?

- [ ] Make the new message box more prominent.

- [ ] The default colours should be NCSS colours. See slack for some colour palettes.

I haven't checked but do the send and voice input buttons react to input?
It's important that there's visual feedback for if the message is submitting etc.

What does the title bar look like in a room? You can add a path to the URL to see.

**Comment by cyphar** [comment]
> I agree the desktop interface is very mobile-y. I am not convinced this is entirely bad

Fair enough, I'll gladly admit this is a matter of taste and mine is probably not as good as folks who actually do UI work. :smile_cat:

**Comment by tfpk** [comment]
Okay, now I think we're ready for review. See below for a few samples:

image

image

image

(the theme is controlled by css variables, so should be pretty easy to customize)

image

Any other feedback?

File: client/necsus.css [hunk 1]
```diff
@@ -7,9 +7,73 @@ main {
   margin-bottom: 0 !important;
 }
 
+/*
+* Theme Fixes
+*/
+
+/* LIGHT */t
```
Comment (kennib) [inline client/necsus.css]: The rogue `t` here is breaking the CSS.

**Comment by kennib** [comment]
Once you fix the commented line and the colour of the buttons (which is white on white) I'm happy to merge.

**Comment by cyphar** [comment]
@tfpk Needs a rebase (I cleaned up trailing whitespace in #41), but also one more thing -- while waiting for a bot to respond to a message (which in the case of some student bots may take several seconds), the message box should be cleared and there should be some sort of spinner to indicate that a bod is "thinking" (maybe replacing the "send" button, do whatever you feel is natural).

The old design had the message box grey out while waiting for a bot to respond, but this redesign doesn't really indicate why necsus isn't responding.

**Comment by tfpk** [comment]
Alrighty! This should be ready to go.

One thing -- I'm getting an intermittent issue where sending to a bot then gets a timeout (even though the bot responds). I don't think this is client side, though -- did anyone else happen to make a change that might cause this?

**Comment by cyphar** [comment]
Thanks to #41 there is now a timeout of 13 seconds for requests. Is that not long enough for you? We can make it longer pretty easily, the only issue was that we needed /some/ kind of timeout because we didn't have one last year. I picked 13 seconds because I was feeling lucky.

**Comment by cyphar** [comment]
I'll merge this now, though I think the default theme colours could be a little bit darker and the header margin could be a little bigger.

**Pull request merged**